### PR TITLE
Unlink integrity files when pycurl errors occur

### PR DESCRIPTION
**Pull request by phmccarty**
Setting `curl.FAILONERROR` means that pycurl will raise an error for certain conditions that previously succeeded. Make sure to unlink integrity files for these error cases.

### CODE REVIEW

File: autospec/pkg_integrity.py [hunk 1]
```diff
@@ -223,6 +223,7 @@ def download_file(url, destination):
             curl.perform()
         except pycurl.error as e:
             print(e.args)
+            os.unlink(sign_file)
```
Comment (bryteise) [inline autospec/pkg_integrity.py]: Could we just not put the perform in a try catch? I don't think unlink inside the open block for the same file is very nice =/.

Comment (phmccarty) [reply autospec/pkg_integrity.py]: Sure, I will fix...

Comment (busykai) [reply autospec/pkg_integrity.py]: That's a good point. We still do this below when checking the HTTP code.

**Comment by busykai** [comment]
Thanks, @phmccarty! Much better now.

**Pull request merged**